### DISCUSSION TITLE
Release Build Failure when using -DCMAKE_BUILD_TYPE=Release

### DISCUSSION THREAD

**Discussion by Llamapaca** [assimp/assimp#5610] [Q&A] [answered]
Hi Assimp community, I'm super new and trying to learn more about assimp and build process via Cmake. I have this weird issue when trying to build assimp as a part of my project dependency as it fails to build, because of warnings being treated as errors. What is weird is that if I rerun without `rm -rf ./build/*` step, the error goes away.

Minimal `CMakeLists.txt` to demonstrate the issue:
```
cmake_minimum_required(VERSION 3.4...3.28 FATAL_ERROR)

project(Release CXX)

add_executable(Release main.cpp)

include(FetchContent)
FetchContent_Declare(
	assimp
	GIT_REPOSITORY https://github.com/assimp/assimp.git
	GIT_TAG v5.4.1
)
FetchContent_MakeAvailable(assimp)
target_link_libraries(Release PRIVATE assimp)
add_dependencies(Release assimp)
```
`main.cpp` is just hello world 

and simple `makefile` to just run it:
```
[default]:
	clear
	mkdir -p build 
	rm -rf ./build/*
	cmake -DCMAKE_BUILD_TYPE=Release -S . -B build  --compile-no-warning-as-error
	echo "###############################################"
	cmake --build ./build --parallel 10
	echo "###############################################"
	./build/Release
```

Error message:
```
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 8 is out of the bounds [0, 8] [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
````
Now the question is, am I doing something wrong or there was an issue and it got fixed at some point ? 

P.S. The issue is not reproducible on current master

**Comment by kimkulling** [comment, marked as answer]
This is a known issue. You can track this issue report to get informed when it will be fixed: https://github.com/assimp/assimp/issues/5557
